obsah/data_types.py

```diff
@@ -0,0 +1,69 @@
+"""
+obsah data types to be used to validate user input
+"""
+
+import argparse
+import inspect
+import os
+import sys
+
+
+class BaseType:
+    """
+    Base type class
+    """
+
+    @property
+    def name(self):
+        """
+        the name of the type
+        this is either the class name of, if set, the `type_name` property
+        """
+        if hasattr(self, 'type_name'):
+            return self.type_name
+        return self.__class__.__name__
+
+    def validate(self, string):
+        """
+        validate user input
+        """
+        return string
+
+class Boolean(BaseType):
+    """
+    Boolean type, accepting true/false and 1/0 only
+    """
+    def validate(self, string):
+        if string.lower() in ['true', '1']:
+            return True
+        if string.lower() in ['false', '0']:
+            return False
+        raise ValueError
+
+
+class Path(BaseType):
+    """
+    Path type, accepting anything that looks like a path
+    """
+    def validate(self, string):
+        return string
+
+
+class File(Path):
+    """
+    File type, accepting any existing file
+    """
+    def validate(self, string):
+        if os.path.exists(string) and os.path.isfile(string):
```

Comment:
Is the `exists` needed here? Feels redundant if you also check `isfile`.

Reply:
probably not needed, yeah